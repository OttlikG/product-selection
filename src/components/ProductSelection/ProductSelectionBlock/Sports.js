
import React from 'react'

export default function Sports () {
  return (
    <div className="block sports-block">
      Sports
    </div>
  )
}
