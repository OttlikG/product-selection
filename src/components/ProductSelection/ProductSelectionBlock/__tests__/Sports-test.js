
import React from 'react'
import Sports from '../Sports'

describe('Sports', () => {
  let props, component, inputs

  const chelsea = {
    category: 'Sports',
    id: 'aaa',
    location: 'LONDON',
    product_name: 'Chelsea TV',
    isSelected: true
  }
  const arsenal = {
    category: 'Sports',
    id: 'bbb',
    location: 'LONDON',
    product_name: 'Arsenal TV',
    isSelected: false
  }

  beforeEach(() => {
    props = {
      data: [chelsea, arsenal],
      selectProductById: jest.fn(),
    }

    component = render(<Sports {...props} />)
    inputs = component.findAll('input')
  })

  it('shoud render all item', () => {
    expect(component.find('.sports-list').children.length).toBe(2)
  })

  it('should set checked state if isSelected true', () => {
    expect(inputs[0].props.checked).toBe(true)
    expect(inputs[1].props.checked).toBe(false)
  })

  it('should call action on change', () => {
    inputs[0].props.onChange()
    expect(props.selectProductById).toHaveBeenCalledTimes(1)
    expect(props.selectProductById).toHaveBeenCalledWith('aaa')
  })
})
