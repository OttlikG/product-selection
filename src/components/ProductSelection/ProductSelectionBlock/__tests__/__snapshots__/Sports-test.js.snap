// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sports shoud render all item 1`] = `
<div
  className="block sports-block"
>
  <div>
    Sports
  </div>
  <div
    className="sports-list"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
         
        Chelsea TV
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
         
        Arsenal TV
      </label>
    </div>
  </div>
</div>
`;

exports[`Sports should call action on change 1`] = `
<div
  className="block sports-block"
>
  <div>
    Sports
  </div>
  <div
    className="sports-list"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
         
        Chelsea TV
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
         
        Arsenal TV
      </label>
    </div>
  </div>
</div>
`;

exports[`Sports should set checked state if isSelected true 1`] = `
<div
  className="block sports-block"
>
  <div>
    Sports
  </div>
  <div
    className="sports-list"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
         
        Chelsea TV
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
         
        Arsenal TV
      </label>
    </div>
  </div>
</div>
`;
